src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -799,14 +796,18 @@ AsyncCleaner::gc_cycle_ret AsyncCleaner::do_gc_cycle()
 {
   if (gc_should_trim_alloc()) {
     return gc_trim_alloc(
-    ).handle_error(
+    ).safe_then([this]() {
+      return update_tail_if_needed();
```

Comment:
Can it be implemented in `TransactionManager::submit_transaction_direct()` instead of here?

It doesn't look reasonable to couple `AsyncCleaner` with `CBJournal`, as `CBJournal` already couples the `JournalTrimmer`.